hoomd/md/TwoStepNPTMTK.cc

```diff
@@ -689,6 +693,39 @@ void TwoStepNPTMTK::updatePropagator(Scalar nuxx, Scalar nuxy, Scalar nuxz, Scal
     m_mat_exp_r_int[5] = m_deltaT*exp_r_fac.z*f_r.z;                                   // zz
     }
 
+// Set Flags from 6 element boolean tuple named box_df to integer flag 
+void TwoStepNPTMTK::setFlags(const std::vector<bool>& value)
+    {
+    bool twod = m_sysdef->getNDimensions()==2;
+    int flags = 0;
+    if (value[0])
+        flags |= int(baroFlags::baro_x);
+    if (value[1])
+        flags |= int(baroFlags::baro_y);
+    if (value[2] && !(twod))
+        flags |= int(baroFlags::baro_z);
+    if (value[3])
+        flags |= int(baroFlags::baro_xy);
+    if (value[4] && !(twod))
+        flags |= int(baroFlags::baro_xz);
+    if (value[5] && !(twod)) 
+        flags |= int(baroFlags::baro_yz);
+    m_flags = flags;
+    }
+
+// Get Flags from integer flag to 6 element boolean tuple
+std::vector<bool> TwoStepNPTMTK::getFlags()
+    {
+    std::vector<bool> result;
+    result.push_back(m_flags & baro_yz);
```

Comment:
Have you tested this by setting flags and ensuring that you get back the same flags you set? I am concerned that the flags appear to be pushed onto the result vector in the opposite order of setFlags. It appears that if you set the flags [1, 0, 0, 0, 0, 0], you will get back the flags [0, 0, 0, 0, 0, 1].